src/core/buffers/adaptation/adaptation_buffer.ts

```diff
@@ -201,17 +215,41 @@ export default function AdaptationBuffer<T>(
     representation : Representation
   ) : Observable<IRepresentationBufferEvent<T>> {
     return observableDefer(() => {
+      const bufferGoalRatio = bufferGoalRatioMap[representation.id] || 1;
```

Comment:
might be better to check for `undefined` (or `null`: `== null`). As `0` would mean a totally different thing.